### PR TITLE
greetd.tuigreet: add man page

### LINKED ISSUE #349072: add tuigreet manpage
## Issue description

Tuigreet seems to have a manpage available at https://github.com/apognu/tuigreet/blob/master/contrib/man/tuigreet-1.scd
but `man tuigreet` does nothing on master.

## Technical details

cc maintainer @luc65r

### PR DESCRIPTION
**Pull request by Frontear**
Closes https://github.com/NixOS/nixpkgs/issues/349072

cc @luc65r as maintainer

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by Frontear** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 349180`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>greetd.tuigreet</li>
  </ul>
</details>

File: pkgs/applications/display-managers/greetd/tuigreet.nix [hunk 1]
```diff
@@ -16,6 +18,14 @@ rustPlatform.buildRustPackage rec {
 
   cargoHash = "sha256-RkJjAmZ++4nc/lLh8g0LxGq2DjZGxQEjFOl8Yzx116A=";
 
+  nativeBuildInputs = [ scdoc ];
+
+  postInstall = ''
+    mkdir -p $out/share/man/man1
+
+    scdoc < contrib/man/tuigreet-1.scd > $out/share/man/man1/tuigreet.1
```
Comment (Aleksanaa) [inline pkgs/applications/display-managers/greetd/tuigreet.nix:26]: Better to use `installManpage`, and add `installShellFiles` to `nativeBuildInputs`.

Comment (Frontear) [reply pkgs/applications/display-managers/greetd/tuigreet.nix:26]: Are you saying to use `installManPage` instead of file redirection from `scdoc`?

Comment (Aleksanaa) [reply pkgs/applications/display-managers/greetd/tuigreet.nix:26]: scdoc is still needed, but the output can be installed by `installManpage` to indicate that we are installing manpage here

Comment (Frontear) [reply pkgs/applications/display-managers/greetd/tuigreet.nix]: Changed, please have a look and let me know if this is satisfactory.

**Review by Aleksanaa — APPROVED** [review]

**Comment by teto** [comment]
thank you <3

**Pull request merged**